install/test/helm_test.go

```diff
@@ -5857,6 +5857,9 @@ metadata:
 							if u.GetKind() == kind && u.GetName() == resourceName {
 								a := getFieldFromUnstructured(u, append(prefixPath, "spec", "ttlSecondsAfterFinished")...)
 								Expect(a).To(BeNil())
+								if helmHookDeletePolicy, ok := u.GetAnnotations()["helm.sh/hook-delete-policy"]; ok {
+									Expect(helmHookDeletePolicy).To(ContainSubstring("hook-succeeded"))
+								}
```

Comment:
```suggestion
								if kind == "Job" {
									helmHookDeletePolicy, ok := u.GetAnnotations()["helm.sh/hook-delete-policy"]
									Expect(ok).To(BeTrue())
									Expect(helmHookDeletePolicy).To(ContainSubstring("hook-succeeded"))
								}
```

It's good to be more specific about our expectations here
We want to explicitly assert that there _is_ a `hook-delete-policy` on the `gateway-certgen` and `gloo-mtls-certgen` jobs